lightning/src/util/ser_macros.rs

```diff
@@ -1102,7 +1106,14 @@ macro_rules! impl_writeable_tlv_based_enum_upgradable {
 					$($($tuple_variant_id => {
 						Ok(Some($st::$tuple_variant_name(Readable::read(reader)?)))
 					}),*)*
-					_ if id % 2 == 1 => Ok(None),
+					_ if id % 2 == 1 => {
```

Comment:
It looks like `ClosureReason` uses this macro with several odd tuple variants, including some added recently: https://github.com/lightningdevkit/rust-lightning/commit/58f7942e11d7a500a605c9ae968855285cadefa7 would this change break downgrades if those variants were written?